test/unit-test/mqtt_agent_utest.c

```diff
@@ -81,14 +136,307 @@ int suiteTearDown( int numFailures )
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+
+void test_MQTTAgent_ResumeSession_failed_publish( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    AgentMessageInterface_t msgCtx;
+    MQTTFixedBuffer_t networkBuffer;
+    TransportInterface_t transportInterface;
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    Command_t command;
+    MQTTPublishInfo_t args;
+    AckInfo_t ackInfo;
+
+    msgCtx.pMsgCtx = &msg;
+    msgCtx.send = send;
+    msgCtx.recv = receive;
+    msgCtx.releaseCommand = commandRelease;
+    msgCtx.getCommand = getCommand;
+
+
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+
+
+    command.pArgs = &args;
+    mqttAgentContext.mqttContext.nextPacketId = 1;
+    ackInfo.packetId = 1;
+    ackInfo.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+    /* Check that failed resending publish return MQTTBadParameter */
+    MQTT_PublishToResend_IgnoreAndReturn( 1 );
+    MQTT_Publish_IgnoreAndReturn( MQTTBadParameter );
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+}
+
+
+
+void test_MQTTAgent_ResumeSession_ack_packetId_not_match( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    AgentMessageInterface_t msgCtx;
+    MQTTFixedBuffer_t networkBuffer;
+    TransportInterface_t transportInterface;
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    Command_t command;
+    MQTTPublishInfo_t args;
+    AckInfo_t ackInfo;
+
+    msgCtx.pMsgCtx = &msg;
+    msgCtx.send = send;
+    msgCtx.recv = receive;
+    msgCtx.releaseCommand = commandRelease;
+    msgCtx.getCommand = getCommand;
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+
+
+    command.pArgs = &args;
+    mqttAgentContext.mqttContext.nextPacketId = 1;
+    ackInfo.packetId = 1;
+    ackInfo.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+
+    /* Check that publish ack packet id matches the next packet id in the mqttContext. */
+    MQTT_PublishToResend_IgnoreAndReturn( 2 );
+    MQTT_PublishToResend_IgnoreAndReturn( MQTT_PACKET_ID_INVALID );
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+
+
+void test_MQTTAgent_ResumeSession_publish_resend_success( void )
```

Comment:
The code path tested here is identical to `test_MQTTAgent_ResumeSession_session_present_with_packetId_Invalid`